core/controllers/editor.py

```diff
@@ -775,3 +777,51 @@ def get(self, exploration_id):
         self.render_json({
             'unresolved_answers': unresolved_answers_with_frequency
         })
+
+
+class LearnerAnswerInfoHandler(EditorHandler):
+    """Handles the learner answer info for an exploration state."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_edit_exploration
+    def get(self, exploration_id):
+        """Handles the GET requests for for learner answer info for an
+        exploration state.
+        """
+        if not constants.ENABLE_SOLICIT_ANSWER_DETAILS_FEATURE:
+            raise self.PageNotFoundException
+
+        state_name = self.request.get('state_name')
+        if not state_name:
+            raise self.PageNotFoundException
+
+        entity_type = feconf.ENTITY_TYPE_EXPLORATION
+        state_reference = (
+            stats_models.LearnerAnswerDetailsModel
+            .get_state_reference_for_exploration(exploration_id, state_name))
+        learner_answer_details = stats_services.get_learner_answer_details(
+            entity_type, state_reference)
+        learner_answer_info_dict_list = [
+            learner_answer_info.to_dict() for learner_answer_info in
+            learner_answer_details.learner_answer_info_list]
+        self.render_json({
+            'learner_answer_info_dict_list': learner_answer_info_dict_list
+        })
+
+    @acl_decorators.can_edit_exploration
+    def delete(self, exploration_id):
+        """Deletes the learner answer info by the given id."""
+
+        if not constants.ENABLE_SOLICIT_ANSWER_DETAILS_FEATURE:
+            raise self.PageNotFoundException
+
+        state_name = self.request.get('state_name')
+        learner_answer_info_id = self.request.get('learner_answer_info_id')
+        entity_type = feconf.ENTITY_TYPE_EXPLORATION
+        state_reference = (
+            stats_models.LearnerAnswerDetailsModel
```

Comment:
Let's handle this in the service layer!

Reply:
I checked that at many places we have used models functions in the controllers file. By the way its just generating state reference.

Reply:
Can we make `delete_learner_answer_info` accept `exploration_id, state_name` as aparameter and generate state_reference internally?

Reply:
Try to handle this in the service layer please.

Reply:
Done

Reply:
is this done? (I'm leaving review here as of now, can you please go through the existing comments and make sure they are resolved?)